File: serde-generate/runtime/java/com/novi/bincode/BincodeDeserializer.java
// Copyright (c) Facebook, Inc. and its affiliates
// SPDX-License-Identifier: MIT OR Apache-2.0

package com.novi.bincode;

import com.novi.serde.DeserializationError;
import com.novi.serde.Slice;
import com.novi.serde.BinaryDeserializer;

public class BincodeDeserializer extends BinaryDeserializer {
    public BincodeDeserializer(byte[] input) {
        super(input);
    }

    public long deserialize_len() throws DeserializationError {
        long value = getLong();
        if (value < 0 || value > Integer.MAX_VALUE) {
            throw new DeserializationError("Incorrect length value");
        }
        return value;
    }

    public int deserialize_variant_index() throws DeserializationError {
        return getInt();
    }

    public void check_that_key_slices_are_increasing(Slice key1, Slice key2) throws DeserializationError {
        // Not required by the format.
    }
}

File: serde-generate/runtime/dart/org/starcoin/bincode/BincodeDeserializer.dart
// Copyright (c) Facebook, Inc. and its affiliates
// SPDX-License-Identifier: MIT OR Apache-2.0

import '../serde/BinaryDeserializer.dart';
import '../serde/Slice.dart';
import 'dart:typed_data';

class BincodeDeserializer extends BinaryDeserializer {
  BincodeDeserializer(Uint8List input) : super(input) {}

  int deserialize_len() {
    return deserialize_u64();
  }

  int deserialize_variant_index() {
    return deserialize_u64();
  }

  void check_that_key_slices_are_increasing(Slice key1, Slice key2) {
    // Not required by the format.
  }
}
